packages/edit-site/src/components/page-patterns/style.scss

```diff
@@ -255,3 +255,31 @@
 		z-index: 2;
 	}
 }
+
+// TODO: this is duplicated from `patterns-menu-items__convert-modal` styles,
+// except for the `z-index`. Need to check if this is still needed.
```

Comment:
I believe the CSS for the modal was required to both constrain the width of the modal and prevent a subtle layout shift when interacting with the category selection control. The rest I think was to bring the styling of the category selector up to the latest approach.

Quickly disabling those styles and interacting with the modal resulted in this:

<img width="1072" alt="Screenshot 2024-01-07 at 1 36 17 pm" src="https://github.com/WordPress/gutenberg/assets/60436221/a2fe1399-27ac-46d6-8bdb-f24ac2bae9a2">